Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -169,14 +179,28 @@ class TIFFLoadingContext {
         BigEndian,
     };
 
-    static ErrorOr<u8> read_component(BigEndianInputBitStream& stream, u8 bits)
+    ErrorOr<u8> read_component(BigEndianInputBitStream& stream, u8 component_index) const
     {
         // FIXME: This function truncates everything to 8-bits
-        auto const value = TRY(stream.read_bits<u32>(bits));
 
-        if (bits > 8)
-            return value >> (bits - 8);
-        return NumericLimits<u8>::max() * value / ((1 << bits) - 1);
+        auto const bits = (*m_metadata.bits_per_sample())[component_index];
+        auto value = TRY(stream.read_bits<u32>(bits));
+
+        switch (m_sample_format[component_index]) {
+        case SampleFormat::Unsigned:
+            if (bits > 8)
+                return value >> (bits - 8);
+            return NumericLimits<u8>::max() * value / ((1 << bits) - 1);
+        case SampleFormat::Float:
+            if (bits == 32) {
+                if (m_byte_order == ByteOrder::LittleEndian)
+                    value = __builtin_bswap32(value);
+                return static_cast<u8>(bit_cast<float>(value));
```

Comment:
1. I'm surprised that floats aren't from 0.0 to 1.0 per channel. The spec says "The default for SMinSampleValue and SMaxSampleValue is the full range of the
data type." What does that even mean for floats?

2. clamp().

4. round_to()?

Reply:
I didn't think too much about it and assumed that it was in the 0-255 range when I saw that this was the value stored in the file.

However, these two fields are not mandatory (and not present in the test image) and the spec also says:
> The presence of these fields makes it possible for readers to assume that data samples are bound to the range [SMinSampleValue, SMaxSampleValue] without scanning the image data.

Which makes me understand that you need to scan the image to deduce the correct range. But I'm not refactoring the whole decoder to support that crap. No wonder why many apps refuse this image.

I will add both tag to the test image, fix the code accordingly, and let's pretend the default case doesn't exist.